### PR TITLE
Add public API to update layout

### PR DESCRIPTION
**Pull request by rayqwer**

### CODE REVIEW

**Comment by rayqwer** [comment]
@anthony-redFox  , @null0rUndefined 

please review

File: lib/GridLayout.ts [hunk 1]
```diff
@@ -746,6 +709,73 @@ export default class GridLayout extends HTMLElement {
...
+  }
+
+  /**
+   * Applies changed to Grid Element wiht passed if
+   * And then recalculates all positions of Grid Elements
+   * @param  {String} i ID of the child.
+   * @param  {Object} changes Changes to apply to child.
+   */
+  changeGridElement(i: string, changes: Partial<GridLayoutElementData>) {
+    const layoutItem = getLayoutItem(this.state.layout, i);
```
Comment (rayqwer) [inline lib/GridLayout.ts]: Check if item is not found

File: lib/GridLayout.ts [hunk 2]
```diff
@@ -746,6 +709,76 @@ export default class GridLayout extends HTMLElement {
     this.syncWithNode(layout);
   }
 
+  #collectLayoutState(): GridLayoutState["layout"] {
+    const layout: GridLayoutState["layout"] = [];
+    [...this.children].forEach((node) => {
```
Comment (anthony-redFox) [inline lib/GridLayout.ts]: Why copy children?

Comment (anthony-redFox) [reply lib/GridLayout.ts]: and we need slot which contains only widgets nodes

Comment (rayqwer) [reply lib/GridLayout.ts]: > Why copy children?

`children` doesn't have method `forEach`. Replaced with `for ... of`



> and we need slot which contains only widgets nodes

Added check for `instance of GridLayoutElement`

File: lib/GridLayout.ts [hunk 3]
```diff
@@ -746,6 +709,77 @@ export default class GridLayout extends HTMLElement {
...
+   * @param  {String} id ID of the child.
+   * @param  {Object} changes Changes to apply to child.
+   */
+  changeGridElement(id: string, changes: Partial<GridLayoutElementData>) {
+    const layoutItem = getLayoutItem(this.state.layout, id);
+    if (!layoutItem) {
+      return;
+    }
+
+    Object.assign(layoutItem, changes);
```
Comment (anthony-redFox) [inline lib/GridLayout.ts]: do not change object by link, create new one and set as new Object

**Pull request merged**